lib/associations/belongs-to-many.js

```diff
@@ -34,18 +28,11 @@ module.exports = (function() {
       this.isSelfAssociation ? (this.as || this.target.tableName) : this.target.tableName
     );
 
-    /*
-     * Normalize through
-     */
-    if (this.through === undefined) {
-      this.through = true;
+    if (this.through === undefined || this.through === true || this.through === null) {
```

Comment:
Perhaps it's better to check the other way around (not a model and not a string) - right now you could pass an int, and this wouldn't fail ;)